core/src/main/java/com/linecorp/armeria/common/HttpRequest.java

```diff
@@ -355,10 +356,54 @@ default HttpRequest authority(String authority) {
      * the trailing headers of the request is received fully.
      */
     default CompletableFuture<AggregatedHttpMessage> aggregate(EventExecutor executor) {
+        requireNonNull(executor, "executor");
         final CompletableFuture<AggregatedHttpMessage> future = new CompletableFuture<>();
-        final HttpRequestAggregator aggregator = new HttpRequestAggregator(this, future);
+        final HttpRequestAggregator aggregator = new HttpRequestAggregator(this, future, null);
         completionFuture().whenCompleteAsync(aggregator, executor);
         subscribe(aggregator, executor);
         return future;
     }
+
+    /**
+     * Aggregates this request. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailing headers of the request is received fully. {@link AggregatedHttpMessage#content()} will
+     * return a pooled object, and the caller must ensure to release it. If you don't know what this means,
+     * use {@link HttpResponse#aggregate()}.
+     */
+    default CompletableFuture<AggregatedHttpMessage> aggregateWithPooledObjects(ByteBufAllocator alloc) {
```

Comment:
I wonder if it's a good (or bad) idea to rename this method to `aggregate()`. Thoughts? Should `StreamMessage.subscribe(..., boolean)` be replaced with `subscribeWithPooledObjects()`?

Reply:
I went back and forth on this many times actually, and had the same question about `subscribe`. Here, the `alloc` is needed for efficient aggregation, and using a `@Nullable` alloc to control pooled or not seems quite unclear and possibly dangerous (when it's a parameter passing in an alloc is almost always the right thing to do since it normally indicates a safe optimization, but not here).

`subscribe` doesn't have the same semantic limitation, though renaming it might make sense for consistency